crates/consensus/src/transaction/typed.rs

```diff
@@ -0,0 +1,256 @@
+            Self::Legacy(tx) => tx.set_chain_id(chain_id),
+            Self::Eip2930(tx) => tx.set_chain_id(chain_id),
+            Self::Eip1559(tx) => tx.set_chain_id(chain_id),
+        }
+    }
+}
+
+impl Encodable for TypedTransaction {
+    fn encode(&self, out: &mut dyn alloy_rlp::BufMut) {
+        match self {
+            Self::Legacy(tx) => tx.encode(out),
+            Self::Eip2930(tx) => tx.encode(out),
+            Self::Eip1559(tx) => tx.encode(out),
+        }
+    }
+}
+
+impl Transaction for TypedTransaction {
+    type Signature = Signature;
+
+    fn encode_for_signing(&self, out: &mut dyn alloy_rlp::BufMut) {
+        match self {
+            Self::Legacy(tx) => tx.encode_for_signing(out),
+            Self::Eip2930(tx) => tx.encode_for_signing(out),
+            Self::Eip1559(tx) => tx.encode_for_signing(out),
+        }
+    }
+
+    fn chain_id(&self) -> Option<alloy_primitives::ChainId> {
+        match self {
+            Self::Legacy(tx) => tx.chain_id(),
+            Self::Eip2930(tx) => tx.chain_id(),
+            Self::Eip1559(tx) => tx.chain_id(),
+        }
+    }
+
+    fn decode_signed(_buf: &mut &[u8]) -> alloy_rlp::Result<alloy_network::Signed<Self>>
+    where
+        Self: Sized,
+    {
+        unimplemented!();
+    }
+
+    fn encode_signed(&self, signature: &Signature, out: &mut dyn alloy_rlp::BufMut) {
+        match self {
+            Self::Legacy(tx) => tx.encode_signed(signature, out),
+            Self::Eip2930(tx) => tx.encode_signed(signature, out),
+            Self::Eip1559(tx) => tx.encode_signed(signature, out),
+        }
+    }
+
+    fn encoded_for_signing(&self) -> Vec<u8> {
+        match self {
+            Self::Legacy(tx) => tx.encoded_for_signing(),
+            Self::Eip2930(tx) => tx.encoded_for_signing(),
+            Self::Eip1559(tx) => tx.encoded_for_signing(),
+        }
+    }
+
+    fn gas_limit(&self) -> u64 {
+        match self {
+            Self::Legacy(tx) => tx.gas_limit(),
+            Self::Eip2930(tx) => tx.gas_limit(),
+            Self::Eip1559(tx) => tx.gas_limit(),
+        }
+    }
+
+    fn gas_price(&self) -> Option<alloy_primitives::U256> {
+        match self {
+            Self::Legacy(tx) => tx.gas_price(),
+            Self::Eip2930(tx) => tx.gas_price(),
+            Self::Eip1559(tx) => tx.gas_price(),
+        }
+    }
+
+    fn input(&self) -> &[u8] {
+        match self {
+            Self::Legacy(tx) => tx.input(),
+            Self::Eip2930(tx) => tx.input(),
+            Self::Eip1559(tx) => tx.input(),
+        }
+    }
+
+    fn input_mut(&mut self) -> &mut alloy_primitives::Bytes {
+        match self {
+            Self::Legacy(tx) => tx.input_mut(),
+            Self::Eip2930(tx) => tx.input_mut(),
+            Self::Eip1559(tx) => tx.input_mut(),
+        }
+    }
+
+    fn into_signed(self, _signature: Signature) -> alloy_network::Signed<Self, Self::Signature>
+    where
+        Self: Sized,
+    {
+        unimplemented!();
+    }
```

Comment:
it is, but it is probably not what you want, right?  As this would return a `Signed<TypedTransaction>`, and what you actually want to do is extract the actual variant contained, use `into_signed` on it and get a `Signed<TxLegacy/TxEip2930/TxEip1559>`.